force-app/main/default/classes/HiringTeamHandler.cls

```diff
@@ -0,0 +1,94 @@
+public with sharing class HiringTeamHandler extends TriggerHandler{
+
+    private List<Hiring_Team__c> newHiringTeam;
+    private List<Hiring_Team__c> oldHiringTeam;
+    private Map<Id, Hiring_Team__c> newHiringTeamMap;
+    private Map<Id, Hiring_Team__c> oldHiringTeamMap;
+    
+    public HiringTeamHandler() {
+
+        this.newHiringTeam = (List<Hiring_Team__c>) Trigger.new;
+        this.oldHiringTeam = (List<Hiring_Team__c>) Trigger.old;
+        this.newHiringTeamMap = (Map<Id, Hiring_Team__c>) Trigger.newMap;
+        this.oldHiringTeamMap = (Map<Id, Hiring_Team__c>) Trigger.oldMap;
+    }
+
+    public override void afterInsert(){
+        List<Job_Application__c> jobAppsToUpdate = setApplicationPrimaryContact(newHiringTeam);
+        List<Hiring_Team__c> hiringTeamsToUpdate = updatePreviousPrimaryContact(newHiringTeam);
+        
+        TriggerHandler.bypass('JobApplicationHandler');
+        update jobAppsToUpdate;
```

Comment:
@triden92 to validate CRUD permissions on DML you can use:

`Database.update(jobAppsToUpdate, AccessLevel.SYSTEM_MODE);`